OsmAnd/src/net/osmand/plus/routing/SuggestionsMapsProvider.java

```diff
@@ -0,0 +1,145 @@
+package net.osmand.plus.routing;
+
+import androidx.annotation.NonNull;
+
+import net.osmand.IndexConstants;
+import net.osmand.Location;
+import net.osmand.data.LatLon;
+import net.osmand.map.WorldRegion;
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.onlinerouting.EngineParameter;
+import net.osmand.plus.onlinerouting.OnlineRoutingHelper;
+import net.osmand.plus.onlinerouting.engine.EngineType;
+import net.osmand.plus.onlinerouting.engine.OnlineRoutingEngine;
+import net.osmand.util.Algorithms;
+
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static net.osmand.util.MapUtils.calculateMidPoint;
+
+public class SuggestionsMapsProvider {
+
+	private static final int DISTANCE = 20000;
+	RouteCalculationParams params;
+	private boolean pointOnWater;
+
+	public SuggestionsMapsProvider(RouteCalculationParams params) {
+		this.params = params;
+	}
+
+	public boolean isPointOnWater() {
+		return pointOnWater;
+	}
+
+	public boolean checkIfObjectDownloaded(String downloadName) {
+		final String regionName = Algorithms.capitalizeFirstLetterAndLowercase(downloadName)
+				+ IndexConstants.BINARY_MAP_INDEX_EXT;
+		final String roadsRegionName = Algorithms.capitalizeFirstLetterAndLowercase(downloadName) + ".road"
+				+ IndexConstants.BINARY_MAP_INDEX_EXT;
+		return params.ctx.getResourceManager().getIndexFileNames().containsKey(regionName) || params.ctx.getResourceManager().getIndexFileNames().containsKey(roadsRegionName);
+	}
+
+	public List<Location> getLocationBasedOnDistanceInterval(List<Location> points) {
+		for (int i = 0; i < points.size(); i++) {
+			int nextIndex = i + 1 < points.size() ? i + 1 : i;
+			while (points.get(i).distanceTo(points.get(nextIndex)) > DISTANCE) {
+				Location location = calculateMidPoint(points.get(i), points.get(nextIndex));
+				points.add(nextIndex, location);
+			}
```

Comment:
1. Original Points are always in place ? It's better to have separate List to return and not to modify original list! 
Especially with method called 'get' (it should not modify anything).
2. if points are distributed like every 50 meters, the method will keep all the points, which is probably too much what's needed